### DISCUSSION TITLE
/articles/sharepoint-framework-optimize-webpack-code-splitting/

### DISCUSSION THREAD

**Discussion by giscus** [Voitanos/voitanos.io-comments#310] [Comments]
Learn how to optimize SharePoint Framework web parts using code splitting techniques with Webpack magic comments for improved performance.
https://www.andrewconnell.com/articles/sharepoint-framework-optimize-webpack-code-splitting/

**Comment by stevebeauge** [comment]
If the imported module is used elsewhere without the magic comment, is it smart enough to still create a chunk? Or, worse, will it create a chunk AND keep the module in the main chunk?

**Reply by andrewconnell** [reply]
No... you need to use the magic comment everywhere you're using it if you use the magic comment approach. It *is* smart enough to not recreate a new chunk as it sees it already has one. so it will just reuse the existing one.

The other approach is to do all this in the webpack configuration object... then you don't need magic comments and you can use regular imports, but that's much harder to implement in SPFx because like I said at the start of the article, you have to manually modify the wbepack configuration object created by the SPFx build toolchain (specifically the **gulp bundle** task).

IMHO, magic comments are easier to work with in SPFx solutions, but if you're using it in lots of places, maybe the webpack config option is a better approach.
